src/guardian/guardian.service.ts

```diff
@@ -53,7 +53,7 @@ export class GuardianService implements OnModuleInit {
         const block = await this.providerService.getBlock();
         const blockHash = block.hash;
 
-        await this.repositoryService.initCachedContracts({ blockHash });
+        await this.repositoryService.initOrWaitCachedContracts({ blockHash });
```

Comment:
Let's return block from `initOrWaitCachedContracts` and use number and hash it in the methods below